src/api/app/models/channel.rb

```diff
@@ -168,7 +168,10 @@ def add_channel_repos_to_project(tpkg, mode = nil)
       next unless mode == :enable_all || !ct.disabled
       # add repositories
       unless cp.project.repositories.find_by_name(repo_name)
-        tpkg.project.add_repository_with_targets(repo_name, ct.repository, [ct.repository])
+        unless tpkg.project.repositories.where(name: repo_name).exists?
+          trepo = tpkg.project.repositories.create(name: repo_name)
+          tpkg.project.add_repository_targets(trepo, ct.repository, [ct.repository])
+        end
```

Comment:
*nitpick*: It would also be very nice if you could use speaking variable names. `tpkg` -> `target_package`, `trepo` -> `target_repository`